sdk/src/builder.rs

```diff
@@ -1422,6 +1418,81 @@ impl Builder {
         Ok(self)
     }
 
+    #[async_generic(async_signature(
+        &mut self,
+        time_authority_url: &str,
+        store: &mut Store,
+        http_resolver: &impl AsyncHttpResolver,
+    ))]
+    fn maybe_add_timestamp(
+        &mut self,
+        time_authority_url: &str,
+        store: &mut Store,
+        http_resolver: &impl SyncHttpResolver,
+    ) -> Result<()> {
+        if self.settings.builder.update_manifest_timestamp_assertion {
+            let provenance_claim = store.provenance_claim().ok_or(Error::ClaimEncoding)?;
+            if provenance_claim.update_manifest() {
+                let parent_claim_id = manifest_label_from_uri(
```

Comment:
This seems like it needs a recursive algorithm to keep searching until we find a non-update manifest.